SQLExecutorComponent/src/main/java/com/astronomvm/sqlExecutor/SQLExecutorComponent.java

```diff
@@ -0,0 +1,110 @@
+        databaseConnectionPropertiesParameterMeta.setType(DataType.DATABASE_CONNECTION_PROPERTIES);
+
+        componentMeta.addParameterMeta(sqlQueryParameterMeta);
+        componentMeta.addParameterMeta(databaseConnectionPropertiesParameterMeta);
+
+        return componentMeta;
+    }
+
+    @Override
+    public void parseInputParameters(Map<String, JSONObject> parametersValues) {
+        this.sqlQuery = parametersValues.get(SQL_QUERY_PARAMETER_NAME).getString("value");
+        this.dcp = new DatabaseConnectionProperties();
+        JSONObject dcpJson = parametersValues.get(DATABASE_CONNECTION_PROPERTIES_PARAMETER_NAME).getJSONObject("value");
+        this.dcp.setDatabaseUrl(dcpJson.getString("databaseUrl"));
+        this.dcp.setJdbcDriver(JDBCDriver.valueOf(dcpJson.getString("jdbcDriver")));
+        this.dcp.setPassword(dcpJson.getString("password"));
+        this.dcp.setUsername(dcpJson.getString("username"));
+        this.dcp.setSchemaName(dcpJson.getString("schemaName"));
+    }
+
+    @Override
+    public ResultFlow execute() throws ComponentException {
+        ResultFlow resultFlow = new ResultFlow();
+
+        try {
+            Class.forName(this.dcp.getJdbcDriver().getClassName());
+            Connection connection = DriverManager.getConnection(this.dcp.getDatabaseUrl(),this.dcp.getUsername(),this.dcp.getPassword());
+            Statement statement = connection.createStatement();
+            boolean isSelectQuery = statement.execute(this.sqlQuery);
+            if(isSelectQuery){
+                ResultSet resultSet = statement.getResultSet();
+                ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+                com.astronomvm.core.model.data.output.ResultSet astronomResultSet = new com.astronomvm.core.model.data.output.ResultSet();
+                RowHeader rowHeader = new RowHeader();
+                astronomResultSet.setRowHeader(rowHeader);
+                IntStream.range(1,resultSetMetaData.getColumnCount()+1).forEach(columnIndex -> {
+                    try {
+                        String columnName = resultSetMetaData.getColumnName(columnIndex);
+                        rowHeader.addColumn(columnName,DataType.STRING);
+                    }catch (SQLException e) {
+                        e.printStackTrace();
+                    }
+                });
+                while (resultSet.next()){
+                    Row row = new Row();
+                    IntStream.range(1,resultSetMetaData.getColumnCount()+1).forEach(columnIndex -> {
+                        try {
+                            Object value = resultSet.getObject(columnIndex);
+                            Column column = new Column();
+                            column.setValue(new AstronomObject(value));
+                            row.addColumn(column);
+                        } catch (SQLException e) {
+                            e.printStackTrace();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarcloud.io/organizations/ahmedriahi-github/rules#rule_key=squid%3AS1148))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=AhmedRiahi_AstronomVM&pullRequest=22&issues=AWZfQr15DZywvPe4LfuP&open=AWZfQr15DZywvPe4LfuP)

[ISSUE_KEY]: AWZfQr15DZywvPe4LfuP